python-frontend/src/test/java/org/sonar/python/tree/IPythonTreeMakerTest.java

```diff
@@ -93,36 +98,43 @@ public void line_magic_statement() {
       "a %= 2\n" +
       "b = a % foo(b)", treeMaker::fileInput);
     statementList = findFirstChildWithKind(parse, Tree.Kind.STATEMENT_LIST);
-    assertThat(statementList.children()).hasSize(4);
-    assertThat(statementList.children().get(0).getKind()).isEqualTo(Tree.Kind.EXPRESSION_STMT);
-    assertThat(statementList.children().get(1).getKind()).isEqualTo(Tree.Kind.LINE_MAGIC_STATEMENT);
-    assertThat(statementList.children().get(2).getKind()).isEqualTo(Tree.Kind.COMPOUND_ASSIGNMENT);
-    assertThat(statementList.children().get(3).getKind()).isEqualTo(Tree.Kind.ASSIGNMENT_STMT);
-    assertThat(findFirstChildWithKind(statementList.children().get(3), Tree.Kind.MODULO)).isNotNull();
+    statements = statementList.children();
+    assertThat(statements).hasSize(4);
+    assertThat(statements.get(0).getKind()).isEqualTo(Tree.Kind.EXPRESSION_STMT);
+    assertThat(statements.get(1).getKind()).isEqualTo(Tree.Kind.LINE_MAGIC_STATEMENT);
+    assertThat(statements.get(2).getKind()).isEqualTo(Tree.Kind.COMPOUND_ASSIGNMENT);
+    assertThat(statements.get(3).getKind()).isEqualTo(Tree.Kind.ASSIGNMENT_STMT);
+    assertThat(findFirstChildWithKind(statements.get(3), Tree.Kind.MODULO)).isNotNull();
 
     parse = parseIPython("print(b)\n" +
-      "%timeit a = foo(b); b = 2\n" +
-      "a += b", treeMaker::fileInput);
+      "%timeit a =\\\n" +
```

Comment:
I don't like that we change existing tests to introduce more complex use cases in them. I think there is value in having multiple tests testing a single functionality (closer to the spirit of a unit test). 

I also think it makes things more difficult to review, especially since there are a lot of minor refactoring in the same commit. Adding a single simple test would make the review trivial, whereas here it's more complex than it should.

Reply:
done, slit up to two tests